### PR TITLE
Implement sessions and games so we can have multiplayer

### LINKED ISSUE #20: Implement sessions and games so we can have multiplayer + add missing tests for some models

### PR DESCRIPTION
**Pull request by RolandoDrRobot**
Closes #20 

The idea is to implement the necessary models to have sessions and games where other can join to your game, this was developed in the game jam and was tested there, still need to implement in unity

Included:
- Tests for the missing models
- The logic for Game and GamePlayer to support multiplayer functionality, including models with their respective tests and traits, and tests for those traits.
- The technical debt includes writing the test for the realms.cairo file, even though all the methods it uses are the traits of those new models. Everything should work, but it's more about covering edge cases and simulating the multiplayer session issue. https://github.com/ByteBuildersLabs/ByteBeastsBackend/issues/30

### Testing and Build
Captura de pantalla 2024-09-07 a la(s) 3 02 14 p  m

### CODE REVIEW

**Comment by coxmars** [comment]
@RolandoDrRobot lgtm, sozo build and sozo test passed. I will review it in more detail tomorrow if you want 🫡

**Comment by danielcdz** [comment]
In the PR we can use Closes #`issue` see this comment: https://github.com/ByteBuildersLabs/ByteBeastsBackend/pull/15#issuecomment-2323823936

**Comment by danielcdz** [comment]
I will review the code in detail tomorrow sir @RolandoDrRobot 🫡

**Review by coxmars — CHANGES_REQUESTED** [review]
Hi @RolandoDrRobot, LGTM just resolve the conflits and we are done 🫡

File: src/models.cairo
Comment (danielcdz) [inline src/models.cairo:1]: Why are we adding this models file after the refactor @coxmars did, which involved splitting the models into separate files?

File: src/models/GamePlayer.cairo
Comment (danielcdz) [inline src/models/GamePlayer.cairo:1]: We don't want to add a test to this model?

File: src/systems/realms.cairo
Comment (danielcdz) [inline src/systems/realms.cairo:1]: I'm not 100% sure but I think we can rename the dojo contract to be different from the actions.cairo one and avoid any issue with the naming, as both contracts are going to have the same name

**Review by coxmars — APPROVED** [review]
Conflicts have been resolved, but I'd like to know the responses to Daniel's feedback 🫡

**Review by coxmars — APPROVED** [review]
LGTM 🫡

**Review by danielcdz — APPROVED** [review]
LGTM

**Pull request merged**